Detectors/TRD/simulation/src/Digitizer.cxx

```diff
@@ -333,6 +325,35 @@ bool Digitizer::convertHits(const int det, const std::vector<HitType>& hits, TRD
       // The distance of the position to the middle of the timebin
       double timeOffset = ((float)timeBinTruncated + 0.5 - timeBinIdeal) / samplingRate;
 
+      // ************************************************************
+      // Check if there are signals already
+      //  - Should we move this block? It's only used here
+      //  - It's only used here...
+      // ************************************************************
+      std::vector<int> pads_in_response(kNpad);
+      std::vector<int> keys_in_response(kNpad);
+      for (int iPad = 0; iPad < kNpad; iPad++) {
+        const int colPos = colE + iPad - 1;
+        pads_in_response[iPad] = colPos;
+        if (colPos < 0) {
+          continue;
+        }
+        if (colPos >= nColMax) {
+          break;
+        }
+        const int key = Digit::calculateKey(det, rowE, colPos);
```

Comment:
so how large is this key space? Is it 2211727? This is not so large.
Instead of the map, we could for instance have a `vector<int> keyToIndex` and a separate `vector<ADCCount> adcs`.
Then if `keyToIndex[key]` gives back the index where in `adcs` the counts are stored or `-1` if it doesn't exist.

This solution would not suffer from logarithm access times but it depends of course on the occupancy (for PbPb it should make sense).

Reply:
The largest key is `2211727`, but there are `1244160` possible keys. We were discussing a lot about these two implementations over the past week. Given that the occupancy of arraADC in the previous implementation (AliRoot) does not go above 10%, even for PbPb (@tdietel, correct me if I am wrong); the map made more sense to us. With the map we had advanced element iteration for other interactions in the digitizer, which also makes sense.

Reply:
Advanced element iteration can be achieved also in the flat lookup version mentioned above. Ultimately we should be interested in the fastest solution but we can try to address this in another iteration.

Reply:
I agree with you; my first reaction was to use a map since the occupancy is low. We can try another iteration (separate PR) after I check the performance of both implementations. Maybe we can list this also in a JIRA.